### PR TITLE
Build: Ensure that production builds are valid ES5

### LINKED ISSUE #13693: Building Tools: add build-time ES5 validation to build
ES6+ code sometimes finds its way into the fallback build, which should only include ES5, when one of our dependencies decides to ship their package with newer syntax.

Calypso added a new build-time check that ensures no non-ES5 syntax is present in the generated fallback build, breaking the build if that's the case. It would be nice to add this to Jetpack as well. This will help us catch these issues as soon as they happen, rather than when an IE11 user reports them.

```
npx eslint --env=es5 --no-eslintrc --no-ignore ./_inc/build ./_inc/blocks
```

-- https://github.com/Automattic/wp-calypso/pull/36596

### PR DESCRIPTION
**Pull request by ockham**
Fixes #13693.

#### Changes proposed in this Pull Request:

Add ES5 validation to production builds.

Inspired by https://github.com/Automattic/wp-calypso/pull/36596.

Can't currently add it to the `extension` build, since there are non-ES5 constructs in them.

#### Jetpack product discussion
Code quality.

#### Does this pull request change what data or activity we track or use?
Nah.

#### Testing instructions:
Run `yarn build-production`, and verify that eslint doesn't report any errors at the end.

#### Proposed changelog entry for your changes:
Add ES5 validation to production builds.

### CODE REVIEW

**Comment by david-szabo97** [comment]
Might try `--no-inline-config`, this flag disables inline eslint comments. I had to do the same for Editing Toolkit since it retains some comments that caused "rules missing" error.

**Comment by david-szabo97** [comment]
How about running it through `terser` before validating? That should remove the comments.
I was experimenting with this at Editing Toolkit
```
npx terser ./editing-toolkit-plugin/*/dist/*.js | npx eslint --parser-options=ecmaVersion:5 --no-eslintrc --no-ignore --stdin
```

**Comment by ockham** [comment]
> How about running it through `terser` before validating? That should remove the comments.
> I was experimenting with this at Editing Toolkit
> 
> ```
> npx terser ./editing-toolkit-plugin/*/dist/*.js | npx eslint --parser-options=ecmaVersion:5 --no-eslintrc --no-ignore --stdin
> ```

Yeah, so running through `terser` is what needs to be done, but I don't want to call it directly in `package.json`. Minification needs to be part of a production build anyway, and if it currently isn't, there's something wrong. `terser` is invoked as part of `calypso-build` -- if it's not run here, then there's either a bug, or we're simply not building in production mode.

(I realize that that means that we might have to make sure we only run validation after a production build; but running it unconditionally from `package.json` might also pose a problem, if we'd run it on an already-minified production build.)

**Comment by david-szabo97** [comment]
No, it's working right actually 🤔 But validation is at the wrong script.
It should be in the `build-production-*` scripts.
```
"build-production": "yarn distclean && yarn install --production=false && yarn build-production-client && yarn build-production-php && NODE_ENV=production yarn build-extensions && NODE_ENV=production yarn build-search && NODE_ENV=production yarn build-packages"
```
Notice how we don't have a `build-production-extensions` script but it's inlined: (same goes for `build-search` and `build-packages`
```
NODE_ENV=production yarn build-extensions
```

Probably we should add a `build-production-extensions` and add validation in that script. (and we can do the same for search and packages)
```
"build-production-extensions": "NODE_ENV=production yarn build-extensions  && yarn validate-es5 -- ./_inc/blocks/"
```

**Comment by ockham** [comment]
Ohh you're right! Looks like I didn't read that `package.json` properly :grimacing:

**Comment by ockham** [comment]
Added validation to the other production builds. The funny part is, those pass validation now, whereas the production build for extensions doesn't :sweat_smile: 

```
```

**Comment by david-szabo97** [comment]
Do you remember the PR that added more modules to the transpile list? Well, that's not happening here.

Take a look at these lines: ( and delete them :) )
https://github.com/Automattic/jetpack/blob/42f04ea1126183f6350d62005b68a38f51bc8a0f/webpack.config.extensions.js#L115-L123

and delete these as well since we aren't using this variable anymore
https://github.com/Automattic/jetpack/blob/42f04ea1126183f6350d62005b68a38f51bc8a0f/webpack.config.extensions.js#L93-L95

That is already fixed in `calypso-build` so we need to remove those lines to make sure we use the `calypso-build`'s config

**Comment by ockham** [comment]
> Do you remember the PR that added more modules to the transpile list? Well, that's not happening here.
> 
> Take a look at these lines: ( and delete them :) )
> https://github.com/Automattic/jetpack/blob/42f04ea1126183f6350d62005b68a38f51bc8a0f/webpack.config.extensions.js#L115-L123
> 
> and delete these as well since we aren't using this variable anymore
> https://github.com/Automattic/jetpack/blob/42f04ea1126183f6350d62005b68a38f51bc8a0f/webpack.config.extensions.js#L93-L95
> 
> That is already fixed in `calypso-build` so we need to remove those lines to make sure we use the `calypso-build`'s config

Great spot, thanks @david-szabo97! Fix coming right up :smile:

**Review by jeherve — COMMENTED** [review]
Is it worth waiting until #16845 gets merged so we can add validation to all builds at once?

File: package.json
Comment (ockham) [reply package.json:40]: Yes! Good spot. Oversight on my part :sweat_smile:

**Comment by ockham** [comment]
> Is it worth waiting until #16845 gets merged so we can add validation to all builds at once?

Yeah, makes sense :+1:

**Comment by ockham** [comment]
@jeherve Since #16845 apparently [requires some more work](https://github.com/Automattic/jetpack/pull/16845#issuecomment-704923725), can we proceed with this one (and maybe add ES5 validation for the extensions build to #16845)?

**Review by jeherve — APPROVED** [review]
> can we proceed with this one

Yeah, sounds like a plan!

**Pull request merged**
